### PR TITLE
[Feature]Support hudi connector

### LINKED ISSUE #7681: Support hudi connector
## Feature request

**Is your feature request related to a problem? Please describe.**

Related with https://github.com/StarRocks/starrocks/issues/5062 #7268 

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by adzfolc**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #7681

### CODE REVIEW

**Comment by adzfolc** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 102 / 174 (58.62%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|19|15.79%|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|28|51|54.90%|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|13|22|59.09%|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|36|56|64.29%|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|13|16|81.25%|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|

**Comment by imay** [comment]
@adzfolc please check sonar alert, have some good advices.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 103 / 175 (58.86%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|19|15.79%|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|29|52|55.77%|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|13|22|59.09%|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|36|56|64.29%|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|13|16|81.25%|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|

**Comment by adzfolc** [comment]
run starrocks_fe_unittest

**Comment by adzfolc** [comment]
run starrocks_fe_unittest

**Comment by adzfolc** [comment]
run starrocks_fe_unittest

**Comment by adzfolc** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 108 / 175 (61.71%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|19|15.79%|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|31|52|59.62%|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|36|56|64.29%|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|16|22|72.73%|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|13|16|81.25%|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|

**Comment by adzfolc** [comment]
run SonarCloud Code Analysis

**Comment by adzfolc** [comment]
Bug alter by SonarCloud won't be really triggered.It can be ignored.

File: fe/fe-core/src/main/java/com/starrocks/catalog/HudiTable.java [hunk 1]
```diff
@@ -165,19 +187,29 @@ public Map<String, HiveColumnStats> getTableLevelColumnStats(List<String> column
 
     @Override
     public void refreshTableCache(String dbName, String tableName) throws DdlException {
-        GlobalStateMgr.getCurrentState().getHiveRepository().refreshTableCache(hmsTableInfo);
+        HiveRepository hiveRepository = GlobalStateMgr.getCurrentState().getHiveRepository();
+        try {
+            if (HiveMetaStoreTableUtils.isInternalCatalog(resourceName)) {
+                hiveRepository.refreshTableCache(hmsTableInfo);
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/catalog/HudiTable.java:193]: 193 and 197. refresh twice?

Comment (adzfolc) [reply fe/fe-core/src/main/java/com/starrocks/catalog/HudiTable.java:193]: > 193 and 197. refresh twice?

deleted

**Review by stephen-shelby — COMMENTED** [review]
LGTM

**Review by stephen-shelby — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by adzfolc** [comment]
run starrocks_fe_unittest

**Comment by adzfolc** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by adzfolc** [comment]
run starrocks_admit_test

**Comment by adzfolc** [comment]
run starrocks_admit_test

**Comment by adzfolc** [comment]
run starrocks_admit_test

**Comment by adzfolc** [comment]
run starrocks_admit_test

**Review by caneGuy — APPROVED** [review]

**Pull request merged**